src/bio2bel/sources/biogrid.py

```diff
@@ -0,0 +1,257 @@
+
+    :param column_name: name of column with multiple values to be simplified
+    :param df: dataframe with multiple values in one column
+    :return: dataframe with expanded values
+    """
+    # identify columns with single values that do not neet to be expanded
+    columns_to_keep = df.columns.to_list()
+    columns_to_keep.remove(column_name)
+
+    # create expanded df
+    lens = [len(item) for item in df[column_name]]
+    df_to_keep = {column: np.repeat(df[column].values, lens) for column in columns_to_keep}
+
+    merged_df = {**df_to_keep, **{column_name: np.concatenate(df[column_name].values)}}
+    return pd.DataFrame(merged_df)
+
+
+def get_processed_biogrid() -> pd.DataFrame:
+    """Load BioGRDID file, filter and rename columns and return a dataframe.
+
+    :return: dataframe of preprocessed BioGRID data
+    """
+    df = _get_my_df()
+    # rename columns
+    df = df.rename(columns=BIOGRID_COLUMN_MAPPER)
+
+    # take relevant columns for source, target, alternative ids, relation and PubMed ID
+    df = df[[SOURCE, TARGET, ALT_SOURCE_ID, ALT_TARGET_ID, RELATION, PUBMED_ID]]
+
+    # filter for uniprot
+    df[SOURCE] = filter_for_prefix_single(list_ids=df[SOURCE], prefix=UNIPROT)
+    df[TARGET] = filter_for_prefix_single(list_ids=df[TARGET], prefix=UNIPROT)
+    # also in alternative ids
+    df[ALT_SOURCE_ID] = filter_for_prefix_multi(list_ids=df[ALT_SOURCE_ID], prefix=UNIPROT)
+    df[ALT_TARGET_ID] = filter_for_prefix_multi(list_ids=df[ALT_TARGET_ID], prefix=UNIPROT)
+
+    # drop rows if no uniprot id
+    df = df.dropna(subset=[SOURCE, ALT_SOURCE_ID], how='all')
+    df = df.dropna(subset=[TARGET, ALT_TARGET_ID], how='all')
+
+    # change uniprot/swiss-prot prefix to uniprot
+    df = df.replace(r'^uniprot/swissprot:.*', r'uniprot:.*', regex=True)
+
+    # expand dataframe if multiple uniprot ids exist
+    df = expand_df(df=df, column_name=ALT_SOURCE_ID)
+    df = expand_df(df=df, column_name=ALT_TARGET_ID)
+
+    # filter for relation
+    df[RELATION] = filter_for_prefix_single(
+        list_ids=df[RELATION],
+        rstrip=')',
+        lstrip='(',
+        separator='"',
+        prefix='(',
+    )
+
+    return df
+
+
+def get_bel() -> BELGraph:
+    """Get a BEL graph for IntAct.
+
+    :return: BEL graph
+    """
+    df = get_processed_biogrid()
+    graph = BELGraph(name=BIOGRID)
+    for _, row in tqdm(df.iterrows(), total=len(df.index), desc=f'mapping {BIOGRID}'):
+        _add_my_row(graph, row)
+    return graph
+
+
+def _add_my_row(graph: BELGraph, row) -> None:  # noqa:C901
+    """Add for every pubmed ID an edge with information about relationship type, source and target.
+
+    :param graph: graph to add edges to
+    :param row: row with metainformation about source, target, relation, pubmed_id
+    :return: None
+    """
+    relation = row[RELATION]
+    source_uniprot_id = row[SOURCE]
+    target_uniprot_id = row[TARGET]
+
+    pubmed_ids = row[PUBMED_ID]
+
```

Comment:
Should be extracting metadata about the "Interaction Detection Method" and adding to the BEL graph.

```bash
cat BIOGRID-ALL-3.5.183.mitab.txt | cut -f 7 | sort | uniq -c
   1 Interaction Detection Method
563446 psi-mi:"MI:0004"(affinity chromatography technology)
137579 psi-mi:"MI:0018"(two hybrid)
1260 psi-mi:"MI:0047"(far western blotting)
3199 psi-mi:"MI:0055"(fluorescent resonance energy transfer)
23305 psi-mi:"MI:0090"(protein complementation assay)
53928 psi-mi:"MI:0096"(pull down)
3352 psi-mi:"MI:0114"(x-ray crystallography)
837735 psi-mi:"MI:0254"(genetic interference)
75116 psi-mi:"MI:0401"(biochemical)
23125 psi-mi:"MI:0415"(enzymatic study)
5219 psi-mi:"MI:0428"(imaging technique)
4969 psi-mi:"MI:0686"(unspecified method)
23152 psi-mi:"MI:1313"(bioid)
```

Reply:
Should I add the Interaction detection method as 'annotation' when adding the edges?